src/GraphQL/Validation/DocumentValidator.cs

```diff
@@ -63,34 +55,27 @@ public class DocumentValidator : IDocumentValidator
         };
 
         /// <inheritdoc/>
-        public async Task<(IValidationResult validationResult, Variables variables)> ValidateAsync(
-            ISchema schema,
-            Document document,
-            VariableDefinitions? variableDefinitions,
-            IEnumerable<IValidationRule>? rules = null,
-            IDictionary<string, object?> userContext = null!,
-            Inputs? variables = null,
-            string? operationName = null)
+        public async Task<(IValidationResult validationResult, Variables variables)> ValidateAsync(ValidationOptions options)
         {
-            schema.Initialize();
+            options.Schema.Initialize();
 
             var context = System.Threading.Interlocked.Exchange(ref _reusableValidationContext, null) ?? new ValidationContext();
-            context.Schema = schema;
-            context.Document = document;
-            context.TypeInfo = new TypeInfo(schema);
-            context.UserContext = userContext;
-            context.Variables = variables;
-            context.OperationName = operationName;
+            context.Schema = options.Schema;
+            context.Document = options.Document;
+            context.TypeInfo = new TypeInfo(options.Schema);
+            context.UserContext = options.UserContext;
+            context.Variables = options.Variables;
+            context.Extensions = options.Extensions;
+            context.OperationName = options.OperationName;
 
+            var rules = options.Rules ?? CoreRules;
             try
             {
                 Variables? variablesObj = null;
 
-                rules ??= CoreRules;
-
                 if (!rules.Any())
                 {
-                    variablesObj = context.GetVariableValues(schema, variableDefinitions, variables ?? Inputs.Empty); // can report errors even without rules enabled
+                    variablesObj = context.GetVariableValues(options.VariableDefinitions); // can report errors even without rules enabled
```

Comment:
If the context holds a reference to the selected Operation, it would not need the variable definitions passed in here either.

Reply:
If an invalid operation name is selected, I think the DocumentExecuter would fail before it even attempts to validate the document (verify?).  And if a validation rule wants the operation name, it could be pulled from Operation.Name (or similar).

Reply:
Yes, already done.